pkg/chaosdaemon/tc_server_test.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chaosdaemon
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	pb "github.com/pingcap/chaos-mesh/pkg/chaosdaemon/pb"
+	"github.com/pingcap/chaos-mesh/pkg/mock"
+)
+
+func Test_daemonServer_AddQdisc(t *testing.T) {
+	g := NewWithT(t)
+
+	defer mock.With("MockContainerdClient", &MockClient{})()
+	c, _ := CreateContainerRuntimeInfoClient(containerRuntimeContainerd)
+	s := &daemonServer{c}
+
+	req := &pb.QdiscRequest{
+		Qdisc: &pb.Qdisc{
+			Type: "netem",
+		},
+		ContainerId: "containerd://container-id",
+	}
+
+	t.Run("should work", func(t *testing.T) {
+		const ignore = true
+		defer mock.With("TcApplyError", ignore)()
+
+		_, err := s.AddQdisc(context.TODO(), req)
+		g.Expect(err).To(BeNil())
+	})
+
+	t.Run("should fail on get pid", func(t *testing.T) {
+		const errorStr = "mock error on Task()"
+		defer mock.With("TaskError", errors.New(errorStr))()
+
+		_, err := s.AddQdisc(context.TODO(), req)
+
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(err.Error()).To(ContainSubstring(errorStr))
+	})
```

Comment:
I recommended writing unit test in table-driven style. 

FYI: https://github.com/golang/go/wiki/TableDrivenTests#parallel-testing

Reply:
I have refactored to table-driven style, but seems like mock not works well with parallel testing, maybe I can open another PR to address parallel testing.